test-components/tcp-sockets/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+wit_bindgen::generate!({
+    world: "reactor",
+});
+
+use helper::{ensure_eq, ensure_ok};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use wasi::{
+    io::poll,
+    sockets::{
+        instance_network,
+        network::{
+            ErrorCode, IpAddressFamily, IpSocketAddress, Ipv4SocketAddress, Ipv6SocketAddress,
+        },
+        tcp_create_socket,
+    },
+};
+
+helper::define_component!(Component);
+
+impl Component {
+    fn main() -> Result<(), String> {
+        let address = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 5001);
+
+        let client = ensure_ok!(tcp_create_socket::create_tcp_socket(IpAddressFamily::Ipv4));
+
+        ensure_ok!(client.start_connect(
+            &instance_network::instance_network(),
+            match address {
+                SocketAddr::V6(address) => {
```

Comment:
This branch won't be hit given the hard-coded IPv4 address above, so we might as well remove it for now.